### PR TITLE
Dshape sanitization issue

### PR DESCRIPTION
**Pull request by cpcloud**

### CODE REVIEW

File: odo/backends/tests/test_sql.py [hunk 1]
```diff
@@ -716,3 +716,12 @@ def test_numeric_append():
...
     assert discover(sa.REAL()) == float32
+
+
+def test_string_dshape_doc_example():
+    x = np.zeros((10, 2))
+    with tmpfile('.db') as fn:
+        t = odo(
+            x, 'sqlite:///%s::x' % fn, dshape='var * {a: float64, b: float64}'
+        )
+        assert [row == (0, 0) for row in t.select().execute().fetchall()]
```
Comment (llllllllll) [inline odo/backends/tests/test_sql.py]: this could be a listof all `False`, maybe `assert all(...)`

Comment (cpcloud) [inline odo/backends/tests/test_sql.py]: ah duh thanks!

File: odo/into.py [hunk 2]
```diff
@@ -1,23 +1,40 @@
...
+
+_into = namespace['_into']
 
 
-@into.register(type, object)
+@curry
+def into(a, b, dshape=None, **kwargs):
+    if isinstance(dshape, (str, unicode)):
+        dshape = datashape.dshape(dshape)
+    assert isinstance(dshape, datashape.DataShape) or dshape is None, \
```
Comment (llllllllll) [inline odo/into.py]: could this be a typeerror? imho typeerror better conveys that the user passed the incorrect type to the function

Comment (cpcloud) [inline odo/into.py]: yep good idea ... a user could still potentially trigger this

**Comment by llllllllll** [comment]
Other than the two comments, lgtm

**Comment by cpcloud** [comment]
gr8 thx for the review

**Pull request merged**